spec/worker_spec.rb

```diff
@@ -88,6 +88,19 @@
       end
     end
 
+    context 'with a custom exception having max: 1 retry' do
```

Comment:
This is a pretty similar setup to L40 (only the job type is different, or did I miss something?)

I have a feeling that adding one new spec is not enough here - if we had a bug in `remaining_retries` and green specs, we should update the existing specs so that they fail, because there are some bad assumptions in their setup. Plus, if I'm not mistaken, the `TestJob` class (from `spec/sample_jobs.rb`) is not used by any of the pre-existing specs. Does this make sense?

Reply:
It's similar, the only difference is that it uses `TestJob` which has a `max: 1` retry count which reproduces the buggy behaviour :+1: